### PR TITLE
Fixes multi-turf objects taking multiplied explosion damage

### PR DESCRIPTION
**Pull request by TobleroneSwordfish**
[BUG] [GAME OBJECTS]
## About the PR 
Fixes multi-turf objects taking explosive damage from every turf they cover. 
Profiler seems to say this doesn't break performance, assuming I'm using it right:
image

## Why's this needed? 
This lead to the flock relay taking around 12 times the explosion damage it should and being one shot by a torpedo.
💀 

## Changelog

```changelog
(u)LeahTheTech
(+)The flock relay now takes the correct amount of explosion damage.
```

### CODE REVIEW

**Comment by Tarmunora** [comment]
Hm. Does this guarantee that the thing will be hit by the single highest-power bit of the explosion? Because I think that's something that should happen for this.

**Comment by Tarmunora** [comment]
> > Hm. Does this (try to) guarantee that the thing will be hit by the single highest-power bit of the explosion? Because I think that's something that should happen for this.
> 
> that is a very good point, I will try to figure out a way of doing that that isn't `O(n)`

🙏  (and if you do, you can also delete some of the special-snowflake code singularites have to deal with that lol)

**Comment by TobleroneSwordfish** [comment]
Okay that was relatively painless, doesn't seem to have performance impacts and `singularity.dm` has lost 4% of its curse value.

**Review by Tarmunora — CHANGES_REQUESTED** [review]
sill a little weirdness where severity and power of an `ex_act()` call might be out of touch with one-another

**Review by Tarmunora — APPROVED** [review]

File: code/datums/controllers/explosion_controls.dm [hunk 1]
```diff
@@ -141,18 +141,21 @@ var/datum/explosion_controller/explosions
...
-			while (queued_explosions.len)
+			while (length(queued_explosions))
 				E = queued_explosions[1]
 				queued_explosions -= E
 				E.explode()
 				next_turf_safe |= E.turf_safe
 
 
+/obj
+	var/datum/explosion/last_explosion = null
```
Comment (ZeWaka) [inline code/datums/controllers/explosion_controls.dm:157]: please clear this in disposing

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/datums/controllers/explosion_controls.dm (hunk 1 above)
Comment (TobleroneSwordfish) [reply code/datums/controllers/explosion_controls.dm:157]: Hopefully this is the right way to do that?

File: code/datums/controllers/explosion_controls.dm [hunk 2]
```diff
@@ -141,18 +141,25 @@ var/datum/explosion_controller/explosions
...
+
+/obj/disposing()
+	src.last_explosion = null
+	..()
```
Comment (ZeWaka) [inline code/datums/controllers/explosion_controls.dm:161]:
```suggestion
	. = ..()
```

File: code/datums/controllers/explosion_controls.dm [hunk 3]
```diff
@@ -141,18 +141,25 @@ var/datum/explosion_controller/explosions
...
-			while (queued_explosions.len)
+			while (length(queued_explosions))
 				E = queued_explosions[1]
 				queued_explosions -= E
 				E.explode()
 				next_turf_safe |= E.turf_safe
 
 
+/obj
+	var/datum/explosion/last_explosion = null
```
Comment (ZeWaka) [inline code/datums/controllers/explosion_controls.dm:157]:
```suggestion
/obj/var/datum/explosion/last_explosion = null
```

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**